crates/noirc_frontend/src/ast/statement.rs

```diff
@@ -14,6 +14,112 @@ use noirc_errors::{Span, Spanned};
 /// for an identifier that already failed to parse.
 pub const ERROR_IDENT: &str = "$error";
 
+/// Ast node for statements in noir. Statements are always within a block { }
+/// of some kind and are terminated via a Semicolon, except if the statement
+/// ends in a block, such as a Statement::Expression containing an if expression.
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub enum Statement {
+    Let(LetStatement),
+    Constrain(ConstrainStatement),
+    Expression(Expression),
+    Assign(AssignStatement),
+    // This is an expression with a trailing semi-colon
```

Comment:
small nit: The comment below ended with a full stop, while this one does not. I do this all the time too and I wonder if there is a nice convention to follow

Reply:
I tend to include the period if the comment contains multiple sentences and exclude it otherwise

Reply:
Yeah that sounds like a good rule :)